src/hotspot/cpu/aarch64/aarch64_neon.ad

```diff
@@ -376,6 +396,39 @@ instruct vcvt2Lto2D(vecX dst, vecX src)
   ins_pipe(pipe_class_default);
 %}
 
+instruct vcvt2Fto2I(vecD dst, vecD src)
+%{
+  predicate(n->as_Vector()->length() == 2 && n->bottom_type()->is_vect()->element_basic_type() == T_INT);
+  match(Set dst (VectorCastF2X src));
+  format %{ "fcvtzv  $dst, T2S, $src\t# convert 2F to 2I vector" %}
+  ins_encode %{
+    __ fcvtzv(as_FloatRegister($dst$$reg), __ T2S, as_FloatRegister($src$$reg));
+  %}
+  ins_pipe(pipe_class_default);
+%}
+
+instruct vcvt4Fto4I(vecX dst, vecX src)
+%{
+  predicate(n->as_Vector()->length() == 4 && n->bottom_type()->is_vect()->element_basic_type() == T_INT);
+  match(Set dst (VectorCastF2X src));
+  format %{ "fcvtzv  $dst, T4S, $src\t# convert 4F to 4I vector" %}
+  ins_encode %{
+    __ fcvtzv(as_FloatRegister($dst$$reg), __ T4S, as_FloatRegister($src$$reg));
+  %}
+  ins_pipe(pipe_class_default);
+%}
+
+instruct vcvt2Dto2L(vecX dst, vecX src)
+%{
+  predicate(n->as_Vector()->length() == 2 && n->bottom_type()->is_vect()->element_basic_type() == T_LONG);
+  match(Set dst (VectorCastD2X src));
+  format %{ "fcvtzv  $dst, T2D, $src\t# convert 2D to 2L vector" %}
+  ins_encode %{
+    __ fcvtzv(as_FloatRegister($dst$$reg), __ T2D, as_FloatRegister($src$$reg));
+  %}
+  ins_pipe(pipe_class_default);
+%}
```

Comment:
These VectorCast patterns should be combined.

Reply:
Sorry, what does "combine" mean. These VectorCast patterns are generated by https://github.com/openjdk/jdk/blob/aa4937cb012185425c852fe582aac41b6a477e77/src/hotspot/cpu/aarch64/aarch64_neon_ad.m4#L256 
the code is already relatively concise.

Reply:
Again, sorry.